File: cs/lc103.cs
public class Solution {
    public IList<IList<int>> ZigzagLevelOrder(TreeNode root) {
        IList<IList<int>> ans = new List<IList<int>>();
        if(root==null)
            return ans;
        Queue<TreeNode> q = new Queue<TreeNode>();
        q.Enqueue(root);
        bool leftToRight = true;
        while(q.Count>0)
        {
            int qCount = q.Count;
            IList<int> curLevel = new List<int>();
            for(int i=0;i<qCount;i++)
            {
                TreeNode cur = q.Dequeue();       
                if(leftToRight)
                {
                    curLevel.Add(cur.val);                    
                }
                else
                {
                    curLevel.Insert(0,cur.val);
                }
                if(cur.left!=null)
                    q.Enqueue(cur.left);
                if(cur.right!=null)
                    q.Enqueue(cur.right);
            }           
            ans.Add(curLevel);
            leftToRight = !leftToRight;
        }
        return ans;
    }
}

File: python/lc103.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def zigzagLevelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        if not root:
            return ans
        d = deque()
        d.append(root)
        reverse = False
        while len(d) > 0:
            cur_list = []
            for i in range(len(d)):
                cur_node = d.popleft()
                cur_list.append(cur_node.val)
                if cur_node.left:
                    d.append(cur_node.left)
                if cur_node.right:
                    d.append(cur_node.right)
            ans.append(cur_list if reverse == False else cur_list[::-1])
            reverse = True if reverse == False else False
        return ans
